azurerm/resource_arm_loadbalancer_outbound_rule_test.go

```diff
@@ -0,0 +1,492 @@
+		return s.Remove("azurerm_lb_outbound_rule.test")
+	}
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMLoadBalancerDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMLoadBalancerOutboundRule_basic(ri, outboundRuleName, testLocation()),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMLoadBalancerExists("azurerm_lb.test", &lb),
+					testCheckAzureRMLoadBalancerOutboundRuleExists(outboundRuleName, &lb),
+					deleteOutboundRuleState,
+				),
+				ExpectNonEmptyPlan: true,
+			},
+			{
+				Config: testAccAzureRMLoadBalancerOutboundRule_basic(ri, outboundRuleName, testLocation()),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMLoadBalancerExists("azurerm_lb.test", &lb),
+					testCheckAzureRMLoadBalancerOutboundRuleExists(outboundRuleName, &lb),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMLoadBalancerOutboundRule_disappears(t *testing.T) {
+	var lb network.LoadBalancer
+	ri := tf.AccRandTimeInt()
+	outboundRuleName := fmt.Sprintf("OutboundRule-%d", ri)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMLoadBalancerDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMLoadBalancerOutboundRule_basic(ri, outboundRuleName, testLocation()),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMLoadBalancerExists("azurerm_lb.test", &lb),
+					testCheckAzureRMLoadBalancerOutboundRuleExists(outboundRuleName, &lb),
+					testCheckAzureRMLoadBalancerOutboundRuleDisappears(outboundRuleName, &lb),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func testCheckAzureRMLoadBalancerOutboundRuleExists(outboundRuleName string, lb *network.LoadBalancer) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		_, _, exists := findLoadBalancerOutboundRuleByName(lb, outboundRuleName)
+		if !exists {
+			return fmt.Errorf("A Load Balancer Outbound Rule with name %q cannot be found.", outboundRuleName)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMLoadBalancerOutboundRuleNotExists(outboundRuleName string, lb *network.LoadBalancer) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		_, _, exists := findLoadBalancerOutboundRuleByName(lb, outboundRuleName)
+		if exists {
+			return fmt.Errorf("A Load Balancer Outbound Rule with name %q has been found.", outboundRuleName)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMLoadBalancerOutboundRuleDisappears(ruleName string, lb *network.LoadBalancer) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		client := testAccProvider.Meta().(*ArmClient).loadBalancerClient
+		ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+		_, i, exists := findLoadBalancerOutboundRuleByName(lb, ruleName)
+		if !exists {
```

Comment:
_minor_ these two lines could be combined:
```
		if _, i, exists := findLoadBalancerOutboundRuleByName(lb, ruleName); !exists {
```

Reply:
since I'm using i later on, I think that I can't combine those two lines.